sqlbase: move all system table descriptions to sqlbase

vivekmenezes opened the pull request:
There were two system tables (RangeEventLogTable, EventLogTable)
that were described outside of sqlbase. This change moves
them to sqlbase together with the other system tables.

Some table descriptors are described as literals in system.go. A
followup PR will eliminate the need to use table descriptor literals
for system tables.

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/9277)

<!-- Reviewable:end -->

dt commented:
Er, this is somewhat the opposite of the direction of some recent work w.r.t `MakeTableDesc` and `sql` vs `sqlbase`.

Creating a TableDescriptor from a SQL statement, thanks to things like INTERLEAVE or FOREIGN KEY, involves looking up and reading other tables, making it generally a poor fit for `sqlbase`.  When `MakeTableDesc` was in `sqlbase`, CreateTable was getting messy trying to work around the fact that `MakeTableDesc` only handled some of the conversion of a CREATE TABLE statement into a TableDesc, attempting to handle some AST nodes in one place and some in another. Moving it to `sql` and was the first step in our plan to unify that and clean it up. While right now it looks like it can easily move back to `sqlbase`, the next step in that process -- which hasn't happened yet -- is adding the FK, INTERLEAVE, etc resolution to it, at which point it will not belong there.

The reason we went with the literals was to avoid the circular dependency that follows from transforming a SQL CREATE statement to a tableDesc might need to look up other tables and looking up tables requires the namespace table to exist -- if creating it doesn't involve  transforming a SQL CREATE statement, then we're fine.

@paperstreet had some followup features for INTERLEAVE that were blocked on this too, and was planning to return to them once `MakeTableDesc` moved to `sql` but then got tied up with backup/restore, but I imagine he might also be concerned about moving it back to `sqlbase`.

---

Review status: 0 of 13 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/9277#-:-KRQDaU1I5jeWkqiORzZ:b76pjxd)*
<!-- Sent from Reviewable.io -->

vivekmenezes commented:
That's good to know. Okay then let me construct a descriptor literal and
redo this change. Coming up.

On Sun, Sep 11, 2016, 6:09 PM David Taylor <notifications@github.com> wrote:

> Er, this is somewhat the opposite of the direction of some recent work
> w.r.t MakeTableDesc and sql vs sqlbase.
>
> Creating a TableDescriptor from a SQL statement, thanks to things like
> INTERLEAVE or FOREIGN KEY, involves looking up and reading other tables,
> making it generally a poor fit for sqlbase. When MakeTableDesc was in
> sqlbase, CreateTable was getting messy trying to work around the fact
> that MakeTableDesc only handled some of the conversion of a CREATE TABLE
> statement into a TableDesc, attempting to handle some AST nodes in one
> place and some in another. Moving it to sql and was the first step in our
> plan to unify that and clean it up. While right now it looks like it can
> easily move back to sqlbase, the next step in that process -- which
> hasn't happened yet -- is adding the FK, INTERLEAVE, etc resolution to it,
> at which point it will not belong there.
>
> The reason we went with the literals was to avoid the circular dependency
> that follows from transforming a SQL CREATE statement to a tableDesc might
> need to look up other tables and looking up tables requires the namespace
> table to exist -- if creating it doesn't involve transforming a SQL CREATE
> statement, then we're fine.
>
> @paperstreet <https://github.com/paperstreet> had some followup features
> for INTERLEAVE that were blocked on this too, and was planning to return to
> them once MakeTableDesc moved to sql but then got tied up with
> backup/restore, but I imagine he might also be concerned about moving it
> back to sqlbase.
> ------------------------------
>
> Review status: 0 of 13 files reviewed at latest revision, all discussions
> resolved, some commit checks failed.
> ------------------------------
>
> *Comments from Reviewable
> <https://reviewable.io:443/reviews/cockroachdb/cockroach/9277#-:-KRQDaU1I5jeWkqiORzZ:b76pjxd>*
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/9277#issuecomment-246208076>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ALOpBB_Vx2BNR6M47AUO_xke5Ucv_KdNks5qpHwtgaJpZM4J6HM3>
> .
>

dt commented:
fwiw, `TestSystemTableLiterals` might be useful -- to get the literals the first time, I copy-pasted a literal descriptor there, along with the SQL I wanted, and let it tell me the descriptor I should have (it prints `%#v` of the proto generated by running MakeTableDesc). i then hand-trimmed a few zero values for readability/personal preference.

---

Review status: 0 of 13 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/9277#-:-KRR95I9mO1BeNAQ58aB:bktv88w)*
<!-- Sent from Reviewable.io -->

vivekmenezes committed:
sqlbase: move all system table descriptions to sqlbase

There were two system tables (RangeEventTable, EventLogTable)
that were described outside of sqlbase. This change moves
them to sqlbase placing them with the other system tables.

vivekmenezes commented:
Ready for a review. Thanks

---

Review status: 0 of 7 files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---

*[config/config_test.go, line 233 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/9277#-KRRWCkcpmZ4XQrl_tVG:-KRRWCkcpmZ4XQrl_tVH:br3c6w) ([raw file](https://github.com/cockroachdb/cockroach/blob/3d2ad3ee105b507cc5e69cbafcd95f3337c95083/config/config_test.go#L233)):*
> ```Go
> 	desc1 := sql.CreateTableDescriptor(reservedStart+1, keys.SystemDatabaseID, "CREATE TABLE system.test1 (i INT PRIMARY KEY)", priv)
> 	schema.AddDescriptor(keys.SystemDatabaseID, &desc1)
> 	desc2 := sql.CreateTableDescriptor(reservedStart+2, keys.SystemDatabaseID, "CREATE TABLE system.test2 (i INT PRIMARY KEY)", priv)
> ```

@mrtracy do you remember why you added these test1 and test2 system tables in this test?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/9277#-:-KRRWfj_VzQpR_D6a7JK:brcqe2z)*
<!-- Sent from Reviewable.io -->

dt commented:
Review status: 0 of 7 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[sql/system_table_test.go, line 120 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/9277#-KRR_hDXochlWwMO8YMu:-KRR_hDXochlWwMO8YMv:bn1j9y8) ([raw file](https://github.com/cockroachdb/cockroach/blob/f48730d4d437c5b901dc8e8cd7b37bb9155fe848/sql/system_table_test.go#L120)):*
> ```Go
> 		if !proto.Equal(&test.pkg, &gen) {
> 			t.Fatalf(
> 				"mismatch between re-generated version and pkg version of %s:\npkg:\n\t%#v\ngenerated\n\t%#v",
> ```

So, one advantage to printing whole thing with `%#v` was you could just copy the output into the file to get it to pass... That said, calling out where it differs is nice too? maybe we could just slip the complete `%#v` of `test.pkg`in on a new line at the end of this message?

---

*[sql/system_table_test.go, line 122 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/9277#-KRRa40AdNptwmOD4cD3:-KRRa40AdNptwmOD4cD4:b-ta8a2k) ([raw file](https://github.com/cockroachdb/cockroach/blob/3d2ad3ee105b507cc5e69cbafcd95f3337c95083/sql/system_table_test.go#L122)):*
> ```Go
> 		gen := sql.CreateTableDescriptor(test.id, keys.SystemDatabaseID, test.schema, sqlbase.NewDefaultPrivilegeDescriptor())
> 		if !proto.Equal(&test.pkg, &gen) {
> 			s1 := fmt.Sprintf("%s", test.pkg)
> ```

should this be `%#v` ?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/9277)*
<!-- Sent from Reviewable.io -->

dt commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> Thanks!

---

Review status: 0 of 7 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/9277#-:-KRRaL9R1nnKNirssoib:b-g2a5fb)*
<!-- Sent from Reviewable.io -->

vivekmenezes commented:
Review status: 0 of 7 files reviewed at latest revision, 3 unresolved discussions.

---

*[sql/system_table_test.go, line 120 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/9277#-KRR_hDXochlWwMO8YMu:-KRTjhyckEq7Q3kif9iq:b-li9bfh) ([raw file](https://github.com/cockroachdb/cockroach/blob/f48730d4d437c5b901dc8e8cd7b37bb9155fe848/sql/system_table_test.go#L120)):*
<details><summary><i>Previously, dt (David Taylor) wrote…</i></summary>
> So, one advantage to printing whole thing with `%#v` was you could just copy the output into the file to get it to pass... That said, calling out where it differs is nice too? maybe we could just slip the complete `%#v` of `test.pkg`in on a new line at the end of this message?
</details>
added the fully generated value at the end of the message separated by a new line

---

*[sql/system_table_test.go, line 122 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/9277#-KRRa40AdNptwmOD4cD3:-KRTjgZg-XGFVdH4XzvK:bb7f2mf) ([raw file](https://github.com/cockroachdb/cockroach/blob/3d2ad3ee105b507cc5e69cbafcd95f3337c95083/sql/system_table_test.go#L122)):*
<details><summary><i>Previously, dt (David Taylor) wrote…</i></summary>
> should this be `%#v` ?
</details>
done

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/9277)*
<!-- Sent from Reviewable.io -->
